weave

```diff
@@ -641,16 +653,20 @@ case "$COMMAND" in
             DNS_ARGS=$(exec_remote dns-args "$@")
             [ "$1" = "--with-dns" ] && shift 1
         fi
-        CIDR=$1
-        shift 1
-        validate_cidr $CIDR
+        CIDRS=""
+        while is_cidr "$1"; do
+            CIDRS="$CIDRS $1"
+            shift 1
+        done
```

Comment:
shift $(collect_cidr_args "$@")

`collect_cidr_args` counts the CIDR parameters and sets `CIDRS` as a side effect (Matthias will love that).